forui/lib/src/widgets/accordion.dart

```diff
@@ -0,0 +1,399 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'dart:math' as math;
+
+import 'package:forui/forui.dart';
+import 'package:forui/src/foundation/tappable.dart';
+import 'package:forui/src/foundation/util.dart';
+import 'package:meta/meta.dart';
+
+/// A controller that stores the expanded state of an [FAccordion].
+class FAccordionController extends ChangeNotifier {
+  late final AnimationController _animation;
+  late final Animation<double> _expand;
+
+  /// Creates a [FAccordionController].
+  FAccordionController({
+    required TickerProvider vsync,
+    duration = const Duration(milliseconds: 500),
+    bool initiallyExpanded = true,
+  }) {
+    _animation = AnimationController(
+      duration: duration,
+      value: initiallyExpanded ? 1.0 : 0.0,
+      vsync: vsync,
+    );
+    _expand = Tween<double>(
+      begin: 0,
+      end: 100,
+    ).animate(
+      CurvedAnimation(
+        curve: Curves.ease,
+        parent: _animation,
+      ),
+    )..addListener(notifyListeners);
```

Comment:
Why do we need to add a listener here?

Reply:
When I remove it, the accordion seems to jump between the states abruptly without any animation.